src/constructor.io/client/ConstructorIO.cs

```diff
@@ -50,6 +59,9 @@ public ConstructorIO(ConstructorioConfig config)
                 options.Add(Constants.SERVICE_URL, config.ServiceUrl);
             }
 
+            // Store the HttpClient instance in options (null means use the static one)
+            options.Add(Constants.HTTP_CLIENT, httpClient);
+
```

Comment:
At this point, would it make sense to: 
1. Perform the checks against the custom `httpClient` to ensure that it's valid
2. Default to the statically defined `httpClient` if no custom client is provided

Since it's a core part of the class instantiation, I think it would make sense to fail here rather than later down the line when making a request is attempted. Resolving which client to use here also means we don't have to resolve it downstream. wdyt?